cmd/deploy/deploy.go

```diff
@@ -618,3 +593,82 @@ func (dc *DeployCommand) cleanUp(ctx context.Context) {
 		oktetoLog.Infof("could not stop local server: %s", err)
 	}
 }
+
+func (dc *DeployCommand) buildImages(ctx context.Context, cfg *corev1.ConfigMap, c kubernetes.Interface, pipelineData *pipeline.CfgData, deployOptions *Options) error {
+	if deployOptions.Build {
+		buildOptions := &types.BuildOptions{
+			EnableStages: true,
+			Manifest:     deployOptions.Manifest,
+			CommandArgs:  deployOptions.servicesToDeploy,
+		}
+		oktetoLog.Debug("force build from manifest definition")
+		if errBuild := dc.Builder.Build(ctx, buildOptions); errBuild != nil {
+			return updateConfigMapStatusError(ctx, cfg, c, pipelineData, errBuild)
+		}
+	} else {
+		stack := deployOptions.Manifest.GetStack()
+		if stack == nil {
+			return nil
+		}
+		stackServices := stack.GetServicesWithBuildSection()
+
+		stackServicesToBuild := setIntersection(stackServices, sliceToSet(deployOptions.servicesToDeploy))
+
+		manifestBuildServices := deployOptions.Manifest.GetBuildServices()
+
+		servicesToBuildSet := setUnion(stackServicesToBuild, manifestBuildServices)
+
+		servicesToBuild := setToSlice(servicesToBuildSet)
+
+		if len(servicesToBuild) != 0 {
+			buildOptions := &types.BuildOptions{
+				EnableStages: true,
+				Manifest:     deployOptions.Manifest,
+				CommandArgs:  servicesToBuild,
+			}
+
+			if errBuild := dc.Builder.Build(ctx, buildOptions); errBuild != nil {
+				return updateConfigMapStatusError(ctx, cfg, c, pipelineData, errBuild)
+			}
+		}
+	}
+
+	return nil
+}
+
+func sliceToSet[T comparable](slice []T) map[T]bool {
```

Comment:
i thought these were moving to the refactor?

Reply:
Since the logic inside `buildImages` is really complicated, I didn't want to do it with a bunch of `for` loops. I think this functions are really simple and aid a lot reading `buildImages` to really understand its logic instead of trying to parse the `go` code